Document/0x05b-Basic-Security_Testing.md

```diff
@@ -207,6 +204,217 @@ Unlike static analysis, dynamic analysis is performed while executing the mobile
+- The app you are testing is verifying if a proxy is set and is not allowing now any communication.
+
+In both scenarios you would need additional steps to finally being able to see the traffic. In the sections below we are describing two possible solutions, ettercap and iptables. You could also use an access point that is under your control to redirect the traffic, but this would require additional hardware and we focus for now on software solutions.
+
+> For both solutions you need to activate "Support invisible proxying" in Burp, in Proxy Tab/Options/Edit Interface.
+
+**iptables**
+
+You can use iptables on the Android device to  redirect all traffic to your interception proxy. The following command would redirect port 80 to your proxy running on port 8080
+
+```bash
+$ iptables -t nat -A OUTPUT -p tcp --dport 80 -j DNAT --to-destination <Your-Proxy-IP>:8080
+```
+
+Verify the iptables settings and check the IP and port.
+
+```bash
+$ iptables -t nat -L
+Chain PREROUTING (policy ACCEPT)
+target     prot opt source               destination
+
+Chain INPUT (policy ACCEPT)
+target     prot opt source               destination
+
+Chain OUTPUT (policy ACCEPT)
+target     prot opt source               destination
+DNAT       tcp  --  anywhere             anywhere             tcp dpt:5288 to:<Your-Proxy-IP>:8080
+
+Chain POSTROUTING (policy ACCEPT)
+target     prot opt source               destination
+
+Chain natctrl_nat_POSTROUTING (0 references)
+target     prot opt source               destination
+
+Chain oem_nat_pre (0 references)
+target     prot opt source               destination
+```
+
+In case you want to reset the iptables configuration you can flush the rules:
+
+```bash
+$ iptables -t nat -F
+```
+
+**Ettercap**
+
+Read the chapter "Testing Network Communication" and the test case "Simulating a Man-in-the-Middle Attack" for further preparation and instructions for running ettercap.
+
+Your testing machine and the Android device must be connected to the same wireless network. Start ettercap with the following command, replacing the IP addresses below with the IP addresses of your Android device and the wireless network's gateway.
+
+```bash
+$ sudo ettercap -T -i en0 -M arp:remote /192.168.0.1// /192.168.0.105//
+```
+
+#### Network Monitoring/Sniffing
+
+ [Remotely sniffing all Android traffic in real-time is possible with tcpdump, netcat (nc), and Wireshark](https://blog.dornea.nu/2015/02/20/android-remote-sniffing-using-tcpdump-nc-and-wireshark/ "Android remote sniffing using Tcpdump, nc and Wireshark"). First, make sure that you have the latest version of [Android tcpdump](https://www.androidtcpdump.com/) on your phone. Here are the [installation steps](https://wladimir-tm4pda.github.io/porting/tcpdump.html "Installing tcpdump"):
+
+```
+# adb root
+# adb remount
+# adb push /wherever/you/put/tcpdump /system/xbin/tcpdump
+```
```

Comment:
please replace with:
\```shell
$ adb root
$ adb remount
$ adb push /wherever/you/put/tcpdump /system/xbin/tcpdump
\```
and make sure that all shell-code is equal in the MSTG 💯  (no slashes ofcourse!)

Reply:
(perhaps replace bash with shell :-D ?

Reply:
done